pkg/util/docker/event_pull.go

```diff
@@ -37,7 +36,9 @@ func (d *DockerUtil) openEventChannel(since, until time.Time, filter map[string]
 		Filters: queryFilter,
 	}
 
-	return d.cli.Events(context.Background(), options)
+	ctx, cancel := context.WithTimeout(context.Background(), d.queryTimeout)
```

Comment:
no cancel?

Reply:
Canceling the requrest context is the user's (`LatestContainerEvents`) responsibility. Changing the method to ask for a `ctx` argument instead, for clearer ownership